linetools/spectralline.py

```diff
@@ -171,64 +182,54 @@ def cut_spec(self, normalize=False, relvel=False):
             else:
                 sig = sig / self.analy['spec'].conti[pix]
 
-        # Velocity
-        self.analy['spec'].velo = self.analy['spec'].relative_vel(
-            self.wrest*(1+self.attrib['z']))
-        # Cut
-        velo = self.analy['spec'].velo[pix] 
-        # Return
+       # Return
         return fx, sig, dict(wave=wave, velo=velo)
 
 
     # EW 
-    def box_ew(self):
+    def measure_ew(self, flg=1):
         """  EW calculation
         Default is simple boxcar integration
-        Observer frame, not rest-frame
+        Observer frame, not rest-frame (use measure_restew below)
           wvlim must be set!
           spec must be set!
 
         Parameters
         ----------
+        flg: int, optional
+          1: Boxcar integration
 
-        Returns:
-          EW, sigEW : EW and error in observer frame
+        Fills:
+        -------
+        self.attrib[ 'EW', 'sigEW' ] : 
+          EW and error in observer frame
         """
+        reload(lau)
         # Cut spectrum
         fx, sig, xdict = self.cut_spec(normalize=True)
         wv = xdict['wave']
 
-        # dwv
-        dwv = wv - np.roll(wv,1)
-        dwv[0] = dwv[1]
-
-
-        # Simple boxcar
-        EW = np.sum( dwv * (1. - fx) ) 
-        varEW = np.sum( dwv**2 * sig**2 )
-        sigEW = np.sqrt(varEW) 
-
+        # Calculate
+        if flg == 1: # Boxcar
+            EW,sigEW = lau.box_ew( (wv, fx, sig) )
+        else:
+            raise ValueError('measure_ew: Not ready for this flag {:d}'.format(flg))
 
         # Fill
         self.attrib['EW'] = EW 
         self.attrib['sigEW'] = sigEW 
 
-        # Return
-        return EW, sigEW
-            
     # EW 
-    def restew(self):
+    def measure_restew(self,**kwargs):
```

Comment:
Maybe 'calc_restew' or 'find_restew'? (because it's shorter.) Same goes for `measure_ew`